e2e_test/source_inline/connection/ddl.slt

```diff
@@ -0,0 +1,115 @@
+# Note: control substitution on will force us to use "\n" instead of "\n" in commands
+control substitution on
+
+# for non-shared source
+statement ok
+set streaming_use_shared_source to false;
+
+statement ok
+create secret sec_broker with (backend = 'meta') as '${RISEDEV_KAFKA_BOOTSTRAP_SERVERS}';
+
+statement error unknown field `foo`
+create connection conn with (type = 'kafka', properties.bootstrap.server = secret sec_broker, foo = 'bar');
+
+statement error Connection type "kinesis" is not supported
+create connection conn with (type = 'kinesis');
+
+statement ok
+create connection conn with (type = 'kafka', properties.bootstrap.server = secret sec_broker, properties.security.protocol = 'plaintext');
+
+query TTT
+select "name", "type_" from rw_catalog.rw_connections;
+----
+conn CONNECTION_TYPE_KAFKA
+
+# unstable test serialization due to iter on hashmap
+# the "connectiond_params" looks like:
+# {"properties.bootstrap.server":"SECRET sec_broker AS TEXT","properties.security.protocol":"plaintext"}
+
+statement error Permission denied: PermissionDenied: secret used by 1 other objects.
+drop secret sec_broker;
+
+statement error expect 'profile' as the key for connection ref, but got: connection
+create table t1 (a int, b varchar) with (
+  connector = 'kafka',
+  connection = connection conn,
```

Comment:
Unlike `secret`, the `connection ` prefix is redundant. 

We use a prefix of `secret` because a value can either be plaintext e.g. `password='123456'` or secret ref `password=secret my_pass`. But for connection, it will always be the connection's identifier.

Reply:
By the way, why we prefer `profile=...` than `connection=...` ?

Reply:
No concrete reason. Just stick to the RFC. The syntax seems okay in the meeting.

Reply:
Acceptable, but I still feel `connection = connection <conn_name>` is redundant. `connection = <conn_name>` looks more elegant to me :)